Update FAQs (#7181)
aws-amplify/docs: src/pages/[platform]/how-amplify-works/faq/index.mdx

Amplify Gen 2 supports a wide range of programming languages for client-side development. This includes dedicated client-side libraries for JavaScript, TypeScript, Dart, Java, Kotlin, and Swift. For backend development, Amplify Gen 2 uses TypeScript.

**In Gen 1, Amplify offered a set of use case categories for building applications (for example, Authentication, Analytics, API, DataStore, Geo, and Predictions). Are those same categories available in Gen 2?**

Amplify Gen 2 offers built-in support for Auth, Data, Storage, and Functions. Other use cases can be implemented in Amplify Gen 2 as well using AWS Cloud Development Kit (AWS CDK) constructs which there is documentation for under the respective category name.

**What if we want to add a feature like AI/ML or Amazon Location Service to our application in Gen 2?**

Because Amplify builds on the AWS CDK, any of the 175+ AWS services can be added to your app using [custom resources](/[platform]/build-a-backend/add-aws-services/custom-resources/) and L2/L1 AWS CDK constructs.

**What happens once my application grows too big and I want to do more configuration with my application (add more features, other AWS services, etc.)?*

> Amplify is built on top of the AWS CDK, so you can add any [AWS service to your Amplify app](/[platform]/build-a-backend/add-aws-services/custom-resources/). You can also override [Amplify-generated configuration of your resources](/[platform]/build-a-backend/add-aws-services/custom-resources/) using the CDK. We also offer standalone L3 CDK constructs for Amplify's Auth, Data, and Storage use cases that you can use within any existing CDK application.

Reviewer: Think you need to assert the answer here is "You can stay with Amplify no matter how big your application grows". Points to be made (please improve language): 1/ Amplify is layered on CDK and CloudFormation, providing standard convention for working with AWS services 2/ You can use any deployment pipeline of your choice if you want more control over your CI.

Author: Updated

}

**Is Gen 2 backward compatible with existing Amplify projects on Gen 1?**

> No, Gen 2 is not backward compatible with existing Amplify projects on Gen 1. Amplify Gen 1 and Gen 2 follow different architectural and tooling paradigms in order for Gen 2 to address common feedback customers had about Gen 1. The two versions are not compatible.

Reviewer 2: I'm not a fan of the way this is answered because it leaves a lot of room for interpretation as to what we mean with "backwards compatible"

Reviewer 3: Maybe something like, `Can I combine a Gen 2 project with Gen 1?` or `Can I add add additional functionality using Gen 2 to my existing Gen 1 project?` ?

Author: **If I have a Gen 1 app, can I use Gen 2 with it?**

Amplify Gen 1 and Gen 2 follow different architectural and tooling paradigms, which was necessary to address  common customer feedback from Gen 1. You will need to use our upcoming migration tooling to move from a Gen 1 to Gen 2 app. You cannot use Amplify Gen 1 (Studio/CLI) in the same app as Gen 2.

> **Is there a way to upgrade an existing Amplify project from Gen 1 to Gen 2?**

Reviewer 2: I'd actually put this as the first question

Author: done

We are working on supporting migrations from Gen 1 to Gen 2. In the meantime, Gen 1 is still fully supported and you can continue to use it in apps you've already created.

**Should I use Amplify Gen 1 or Gen 2 in new apps?**

If you're building a new app, we recommend you use Amplify Gen 2.

**Does Amplify Gen 2 support DataStore?**

> Amplify code-first DX (Gen 2) supports GraphQL APIs without DataStore. We will release migration support for moving DataStore Gen 1 apps to Gen 2.

Reviewer 2: ```suggestion
Amplify Gen 2 supports GraphQL APIs without DataStore. We will release migration support for moving DataStore Gen 1 apps to Gen 2.
```

**What programming languages does Amplify Gen 2 support?**

Amplify Gen 2 supports a wide range of programming languages for client-side development. This includes dedicated client-side libraries for JavaScript, TypeScript, Dart, Java, Kotlin, and Swift. For backend development, Amplify Gen 2 uses TypeScript.

**In Gen 1, Amplify offered a set of use case categories for building applications (for example, Authentication, Analytics, API, DataStore, Geo, and Predictions). Are those same categories available in Gen 2?**

> Amplify code-first DX (Gen 2) supports Auth, Data, Storage, and Functions. Other use cases can be implemented using AWS Cloud Development Kit (AWS CDK) constructs which there is documentation for under the respective category name.

Reviewer 2: ```suggestion
Amplify Gen 2 offers built-in support for Auth, Data, Storage, and Functions. Other use cases can be implemented in Amplify Gen 2 as well using AWS Cloud Development Kit (AWS CDK) constructs which there is documentation for under the respective category name.
```

**What if we want to add a feature like AI/ML or Amazon Location Service to our application in Gen 2?**

> Because Amplify builds on the AWS CDK, any of the 175+ AWS services can be added to your app using [custom resources](/[platform]/build-a-backend/add-aws-services/custom-resources/) and L2/L1 AWS CDK constructs.

Reviewer 2: I don't fully get this question. I think it's answered by the previous question.

Author: If you're reading FAQs top to bottom yes, but I think these are two different topics 1/ what happens to the non data/auth/functions/storage categories? and is extensibility possible? I think most FAQs have questions with similar answers but different questions. Plus if someone isn't familiar with CDK they may not know this.

**What happens once my application grows too big and I want to do more configuration with my application (add more features, other AWS services, etc.)?*

> Amplify is built on top of the AWS CDK, so you can add any [AWS service to your Amplify app](/[platform]/build-a-backend/add-aws-services/custom-resources/). You can also override [Amplify-generated configuration of your resources](/[platform]/build-a-backend/add-aws-services/custom-resources/) using the CDK.

Reviewer 2: ```suggestion
Amplify is built on top of the AWS CDK, so you can add any [AWS service to your Amplify app](/[platform]/build-a-backend/add-aws-services/custom-resources/). You can also override [Amplify-generated configuration of your resources](/[platform]/build-a-backend/add-aws-services/custom-resources/) using the CDK. We also offer standalone L3 CDK constructs for Amplify's Auth, Data, and Storage use cases that you can use within any existing CDK application.
```

> Amplify is built on top of the AWS CDK, so you can add any [AWS service to your Amplify app](https://docs.amplify.aws/gen2/build-a-backend/add-aws-services/custom-resources/). You can also override [Amplify-generated configuration of your resources](https://docs.amplify.aws/gen2/build-a-backend/add-aws-services/custom-resources/) using the CDK.

Reviewer 4: Do we need to update these links? It looks like it's pointing to the `gen2` subpath which no longer exists

I think we could do something like this:
```suggestion
Amplify is built on top of the AWS CDK, so you can add any [AWS service to your Amplify app](/[platform]/build-a-backend/add-aws-services/custom-resources/). You can also override [Amplify-generated configuration of your resources](/[platform]/build-a-backend/add-aws-services/custom-resources/) using the CDK.
```

Author: Yep, good catch!

**How much does it cost to operate Amplify Gen2?**

You can read all about Amplify's pricing on our [pricing page](https://aws.amazon.com/amplify/pricing/).

**Can I use any other database storage other than Amazon DynamoDB? Does it have to be noSQL?**

> You can also use an [existing MySQL or PostgreSQL database](/react/build-a-backend/graphqlapi/connect-api-to-existing-database/) in an Amplify app.

Reviewer 4: Sorry @aspittel, just realized, should this link be platform agnostic or should it link to the `react` page?

Author: Updated to [platform]!

> You can also use an [existing MySQL or PostgreSQL database](/[platform]/build-a-backend/graphqlapi/connect-api-to-existing-database/) in an Amplify app.

Reviewer 2: ```suggestion
You can also use an [existing MySQL or PostgreSQL database](/[platform]/build-a-backend/data/connect-api-to-existing-database/) in an Amplify app. With Gen 2's [custom query and mutation support](/[platform]/build-a-backend/data/custom-business-logic), you can integrate with any existing data source through a Lambda function as well. 
```

> You can also use an [existing MySQL or PostgreSQL database](https://docs.amplify.aws/react/build-a-backend/graphqlapi/connect-api-to-existing-database/) in an Amplify app.

Reviewer 4: Could we use this link instead?
```suggestion
You can also use an [existing MySQL or PostgreSQL database](/react/build-a-backend/graphqlapi/connect-api-to-existing-database/) in an Amplify app.
```
